---
layout: page
title: Galaxy at a Glance
summary: "A brief introduction to Galaxy"
sidebar_link: true
categories:
  - Galaxy
tags:
  - Galaxy
  - web framework
---

# A brief introduction to Galaxy

The *Data analysis and interpretation for clinical genomics* on-line course is based on the 
[Galaxy](https://galaxyproject.org/) platform, therefore a basic knowledge of Galaxy is required
to get most out of the course.

The current tutorial will help you to familiarize with the Galaxy interface. It will cover the following topics:

- Description of Galaxy
- How to login in to the server
- Galaxy interface
- Getting data into Galaxy
- Reference genomes, data libraries and workflows

---
[![Galaxy]({{site.url}}{{site.baseurl}}/images/galaxy_logo_25percent_transparent.png)](https://galaxyproject.org/)

- Galaxy is a **web-based** analysis platform for computational biomedical research, designed for biologists to analyze their own data. A default installation comes with most of the popular bioinformatics tools already installed and ready for use. It's developed at Penn State, Johns Hopkins and G. Washington universities with substantial outside contributions
- Galaxy is an **open source** project under [Academic Free License](https://opensource.org/licenses/AFL-3.0)
- It's widely used all around the world, with more than 7,000 [citations](https://galaxyproject.org/galaxy-project/statistics/)
- There are more than 80 public Galaxy servers around the world and some are tailored with specific toolsets for human genomics, microbiology, etc. [public Galaxy servers](https://galaxyproject.org/public-galaxy-servers/)

---
### Core values of Galaxy

- **Accessibility**
  - Users without programming experience can easily specify parameters, run tools, workflows and parse/filter data
- **Reproducibility**
  - Galaxy captures information so that any user can repeat and understand a complete computational analysis
- **Transparency**
  - Users are able to share and publish datasets and analyses via the web and create complex analysis workflows including multiple tools

---

### Log in to the Galaxy server

The course is based on the European Galaxy server running at [https://usegalaxy.eu](https://usegalaxy.eu), which is
partner of this training initiative and it's hosting the tools described in the lectures.
If you already have an account on an alternative server (e.g. [http://usegalaxy.org](http://usegalaxy.org)) you should
open a new one at [https://usegalaxy.eu](https://usegalaxy.eu).
Registration is **free**, and you get access to **250GB** of disk space for your analysis.

1. Open your browser. We recommend Chrome or Firefox (please don't use Internet Explorer or Safari).
1. Go to the European Galaxy server running at [https://usegalaxy.eu](https://usegalaxy.eu).
   - If you have previously registered on this server just log in:
     - On the top menu select: User -> Login
     - Enter your user/password
     - Click Submit
   - If you haven’t registered on this server, you’ll need to do now.
     - On the top menu select: User -> Register
     - Enter your email, choose a password, repeat it and add a one word name (lower case)
     - Click Submit
     
---
### Main Galaxy interface
![Galaxy user interface]({{site.url}}{{site.baseurl}}/images/galaxy_interface.png)

The Galaxy home page is separated into 3 parts, with a menu at the top of the page:
1. The tool list on the left
1. The viewing pane with the tool interface in the middle
1. Analysis and data history on the right

---
### Top menu

![Top menu]({{site.url}}{{site.baseurl}}/images/galaxy_interface_menu.png)

- *Analyze Data* - go back to the 3-panels home page
- *Workflow* - access existing workflows or create new one using the editable diagrammatic pipeline
- *Shared data* - access data libraries, histories, workflows, visualizations and pages shared with you
- *Visualization* - create new track browser and access your saved visualisations
- *Help* - links to Galaxy Biostar (Q&amp;A), Galaxy Community Hub (Wiki), and Interactive Tours
- *User* – your preferences and saved histories, datasets, and pages

---
### Tool interface

![Tool interface]({{site.url}}{{site.baseurl}}/images/101_13.png)

---
### Tools

- Each tool is a text file describing:
  - input datasets, parameters, commands, and outputs
  - help, tests, citations, dependency requirements
- Free tool store: [Galaxy Tool Shed](https://toolshed.g2.bx.psu.edu/)
  - Thousands of tools already available
  - Every software can be embedded
    - If a tool is not available, ask the Galaxy community for help!
  - Only a Galaxy admin can install tools (you'll not be able to install new tools
    on [https://usegalaxy.eu](https://usegalaxy.eu/))
- New versions can be installed without removing old ones to ensure reproducibility

A tool describes:
- the input datasets and their datatypes
- the tool parameters (numerical, text, boolean, selections, colour)
- how to generate a command to execute the tool with the specified inputs and parameters
- the output datasets the tool should produce and their datatypes

---
### History

![History]({{site.url}}{{site.baseurl}}/images/history.png)

- Location of all your analyses 
- Collect all datasets (files) produced by tools you run
- At the heart of Galaxy’s reproducibility. 
  For each dataset, the history tracks:
  - Name, format, size, creation time, datatype-specific metadata
  - Tool id and version, inputs, parameters
  - Standard output (`stdout`) and error (`stderr`)
  - State (grey: waiting; yellow: running; green: success; red: failed)
  - Hidden, deleted, purged (== permanently deleted)

---
### Multiple histories

You can have as many histories as you want
 - Each history should correspond to a different analysis
 - and should have a meaningful name

---
### History options menu

- History behavior is controlled by the *History options* (gear icon)
![History options gear button]({{site.url}}{{site.baseurl}}/images/galaxy_interface_history_menu.png)

- Most options are self explanatory
  - *Create New* history will **not** make your current history disappear
  - To list all your histories, choose *Saved Histories*
  - You can *Copy Datasets* from one history to another

---
# Loading data

### Importing data into Galaxy

There are many ways to get data into Galaxy:
 - Copy/paste from a file
 - Upload data from local file on your computer
 - Upload data from internet by URL
 - Upload data from database queries
   - UCSC, BioMart, ENCODE, modENCODE, Flymine etc.
 - Download shared data from public libraries or shared Data libraries, Histories, Workflows,
   Visualizations, and Pages

It is important to note that Galaxy has the concept of **datatype** built in.
This means that each file stored needs to have its type described to Galaxy as it is being made available.
Examples of file types are: text, fastq, fastq, bam, vcf, bed, tabular etc.

When uploading, datatype can be automatically detected or assigned by user.
- For datasets produced by a tool, the datatype is assigned by the tool
- Tools only accept input datasets with the appropriate datatypes
- You can change the datatype in 2 ways:
  - Edit Attributes -&gt; Datatype (to fix a wrongly assigned datatype)
  - Edit Attributes -&gt; Convert Formats (converts the original dataset)

We will tell Galaxy what type of file each one is as we upload it.

Here we'll focus on the upload from:
 1. a local file
 1. remove file by URL
 1. from the Shared Data libraries

First we start a new history for this tutorial:
 - click on the *+* icon at the top of the History panel.

#### Option 1: Upload a local file from your computer

With this method you can get most of the files on your own computer into Galaxy. The file size should be &lt; 2GB. Galaxy also accepts compressed file and so it will uncompress it during upload.

 - Download the following file to your computer: [CNV_TruSeq_Chr2.bed](https://zenodo.org/record/3531578/files/CNV_TruSeq_Chr2.bed?download=1)
   (To download this file, open the link into a new browser tab. The download should start automatically. If not, right-click on the link and *Save link as...*)
 - From the Galaxy tool panel, click on Get Data -> Upload File
 - Click the *Choose local file* button
 - Find and select the **CNV_TruSeq_Chr2.bed** file you downloaded and click Open
 - Set the "Type" (= file format) to **bed**
 - Click the Start button

The file will now appear in your current history.

#### Option 2: Upload a file from an URL
If a file exists on a web resource and you know its web address, you can directly load it
into Galaxy by providing its URL:

 - From the tool panel, click on Get Data -> Upload File
 - Click on the Paste/Fetch Data button
 - Copy and paste the following URL into the URL/Text box: *https://zenodo.org/record/3531578/files/CNV_TruSeq_Chr2.bed?download=1*
 - Set the file format to **bed**
 - Click Start

#### Option 3:  Get data from a Shared Data Library
Now we are going to get the same file from a shared Data Library.

- Go to the menu at the top of the screen and click *Shared Data* / *Data Libraries*.
- Click on *Galaxy courses*, then ***[Sigu](https://usegalaxy.eu/library/list#folders/F3d08bb711e4e3b26)***

- Click the checkbox next to **CNV_TruSeq_Chr2.bed**
- Click the *To History* button at the top of the page and select *As Datasets*
- Click the "Import" button

Finally, click *Analyse Data* in the menu at the top of the screen to return to your history.
The file will be loaded into your history as a **bed** file.

---
### Reference genomes

Genome build specifies which genome assembly a dataset is associated with, e.g. mm10, hg19, hg38.
Genome build is automatically assigned by software such as sequence aligners based on the
reference genome used as input. Some other tools inherit the genome build from the input file.
For datasets uploaded from external resources (e.g. local file or URL) genome build can be
manually assigned by the user.

To add a missing genome assembly, or to fix a wrongly assigned reference genome,
you can change the genome build as follows:
 - Edit Attributes -&gt; Database/Build

---
### Data Libraries

Data libraries provide a way to conveniently share Galaxy datasets within a group of Galaxy
users or with everybody that has access to a specific instance of Galaxy.
For example, datasets used in this course are available in Library:
*[FIXME NAME OF SHARED DATA LIBRARY]* then *[FIXME NAME OF SUBFOLDER]*

 - The dataset's size does not count towards user's quota.
 - Every dataset in the library is *stored only once* no matter how many users are using it in their histories.
 - It's possible to set roles and groups to control permissions on library/dataset level.
 - Only administrators of the Galaxy server can create *Data libraries*.

---

# Workflows

A workflow is a series of tools that run in sequence as a batch operation:

A workflow can be:
- **extracted** from a history: it allows to easily convert an existing history into an analysis workflow
- **built manually** by adding and configuring tools using the workflow canvas
- **imported** using an existing shared workflow

### Workflow interface

![Workflow interface]({{site.url}}{{site.baseurl}}/images/RNA-Seq_workflow_editing.png)


### Why would you want to create workflows?

- **Re-run** the same analysis on different input data sets
- Change parameters before re-running a similar analysis
- Make use of the workflow job **scheduling**
  - job is submitted as soon as its inputs are ready
- Create **sub**-workflows: a workflow inside another workflow
- Share workflows for publication and with the community

---
### Data sharing

- You can share your Galaxy items - histories, workflows, visualizations, and pages - with
  other people in three different ways:
  - Directly using a Galaxy account's **email addresses** on the same instance
  - Using a **web link**, with anyone who knows the link
  - Using a web link and **publishing** it to make it accessible to everyone from the *Shared Data* menu

- Tools are shared using the free tool store: [Galaxy Tool Shed](https://toolshed.g2.bx.psu.edu/)

---
