---
layout: page
title: Quality control
summary: "Quality control of reads"
sidebar_link: true
---

# Quality control of reads



## Computation of per-base coverage depth at specific genomic intervals

Commercial next-generation sequencing platform usually provide users with analysis programs that include tools for the identification of low coverage regions (for instance, target regions that have a coverage depth lower than 20x).

The present tutorial is aimed to show how to perform a custom coverage analysis of NGS experiment(s) by using tools that are available in Galaxy.
 
Starting material:
* Alignment (*bam*) file(s) on which you want to perform the coverage evaluation.
* A reference *bed* file listing the genomic regions for which you want to obtain coverage data. If you performed a targeted sequencing experiment by using commercial kits (either custom or from the catalogue), you should already have obtained a *bed* file listing the target regions: it should be the file you want to use.

>#### NOTE: BED format specifications 
>
>**BED files** are tab-delimited files with one line for each genomic region.
>The first 3 fields (columns) are required:
>1. chromosome; 
>2. the starting position;
>3. the ending position.
>
>Further columns may be present but are optional.
>
>Further details may be foud here: [UCSC BED format specifications](https://genome.ucsc.edu/FAQ/FAQformat.html#format1)
>
>Please be aware that in BED files the description of genomic regions follows the “0-start, half-open” coordinate system. Further details may be foud here: [the "0-based, half-open" UCSC Genome Browser Coordinate Counting Systems](http://genome.ucsc.edu/blog/the-ucsc-genome-browser-coordinate-counting-systems/). Practically speaking, it means that the starting position is diminished by 1 bp with respect to the actual position. For instance, the region "chr2 50149071 50149399" in a *bed* file corresponds to the genomic interval from position 50149072 (included) to position 50149399 (included) on chromosome 2 (when the first base of the chromosome is defined as position 1). If you want to indicate a single base position in a *bed* file, it should be written like this: "chr2 50149071 50149072".

### 1.Compute per base coverage depth with BEDtools
Before starting, you have to upload the files you need for the analysis, following standard Galaxy procedure. 

You may use files provided as examples with this tutorial and called `Panel_alignment.bam` and `Panel_Target_regions.bed`. 

Please check that uploaded file datatypes (formats) are properly recognized by selecting `edit attributes` (i.e. the pencil sign in correspondence of each file you can find in your history) (indicated by the red arrow in figure 1) and then the tab datatypes (the blue arrow in figure 1). If the datatype is wrong, select the correct one from the drop-down list and press the `change datatype` button (the green arrow n figure 1).

![Figure 1]({{site.url}}{{site.baseurl}}/images/cov_fig1.png)

Once ready, you can select the tool named `bedtools Compute both the depth and breadth of coverage` in the `Operate on Genomic Intervals` section (see the red arrow in figure 2).

1. Select the *bed* file listing the target regions as "file A" (blue arrow in figure 2) and the *bam* file(s) you want to analyze as "file B" (green arrow in figure 2) (they should be listed in the drop-down menu if they have the correct format). You may analyze one or more *bam* files in a single run. 
2. If you want to analyze two or more .bam files, you can further choose if you want all the results in a single file or one output file per input "file B" by selecting the desired option under the `Combined or separate output files` menu.
3. Select "Yes" for the option `Report the depth at each position in each A feature` (yellow arrow in figure 2) and check that all the other options are set to "No".
4. Star (Execute) the analysis.

![Figure 2]({{site.url}}{{site.baseurl}}/images/cov_fig2.png)

Output file, which will be called "coverage_depth.bed" from now on, will contain all the fields of the original target_regions.bed file plus two further columns:
1. the first value indicates a specific base within the reported genomic interval. For instance, if the genomic interval described by the first 3 field is "chr2 50149071 50149072" and the first new field reports the number 1000, it means that the coverage value refers to nucleotide 50150071 (i.e.: 50149071 + 1000) on chromosome 2;
2. the second value indicates the depth of coverage for the defined base.

### 2.Sort files
Since entries in the "coverage_depth.bed" may not be in the desired order, you can sort it by genomic positions.
For this purpose you may want to use the `Sort` tool in the `Text Manipulation` section (check out the blue arrow in figure 3).

You may sequentially sort on different columns:
1. first you can sort by chromosome by selecting column 1 (green arrow in figure 3) in "ascending order" and selecting the "flavor" `Natural/Version sort (-V)`, which allows for sorting chromosomes in their "natural" order (with alphabetical order chr10 will be right after chr1, chr2 after chr19 and so on);
2. after inserting a new column section (red arrow in figure 3), you can sort by column 2 in "ascending order" with `Fast numeric sort (-n)`;
3. after inserting a further column section, you can sort by column 3 in "ascending order" with `Fast numeric sort (-n)`;
4. after inserting a final column section, you can sort by column 3 in "ascending order" with `Fast numeric sort (-n)`.

![Figure 3]({{site.url}}{{site.baseurl}}/images/cov_fig3.png)

The obtained output file, which will be called "sorted_coverage_depth.bed" from now on, will be sorted first by chromosome, then by starting position, by ending position and by the actual position of the specific base in the considered genomic interval.

### 3.Remove duplicate rows
If your file contains duplicated lines you may want to remove them for further processing.
For this purpose you can use the `Unique lines assuming sorted input file` tool in the `Text Manipulation` section.

### 4.Some manipulation of the *bed* file 
You may follow the following steps to manipulate the "sorted_coverage_depth.bed" and to obtain a *bed* file listing the exact base position to which each coverage value is referred.
For instance instead of having "chr2 50149071 50149399 NRXN1 1 2335" in the first row of your file, you will get "chr2 50149071 50149072 NRXN1 2335".

These steps will add further columns at the end of each line defining the base position with the “0-start, half-open” coordinate system.

1. Select the `Compute an expression on every row` tool in the `Text Manipulation` section (indicated by the blue arrow in figure 4);
2. add the following expression "c2+c5-1" to obtain the sum of the values in columns 2 and 5 minus 1 (it will be used as the new starting position in the final file);
3. select the file "sorted_coverage_depth.bed";
4. select "yes" to `round Results?` (green arrow in figure 4);
5. excute;
6. you can rename the output as "temp1_coverage_depth.bed";
7. select the `Compute an expression on every row` tool in the `Text Manipulation` section;
8. add the following expression "c2+c5" to obtain the sum of the values in columns 2 and 5 (it will be used as the new ending position in the final file). You can also use the expression ; 
9. select the file "temp1_coverage_depth.bed";
10. select "yes" to `round Results?`;
11. excute;
12. you can rename the output as "temp2_coverage_depth.bed";
13. select the `Table Compute` tool in the `Text Manipulation` section (indicated by the blue arrow in figure 5);
14. select the file "temp2_coverage_depth.bed";
15. select the option `Drop, keep or duplicate rows and columns` from the drop-down menu `Type of table operation` (indicated by the green arrow in figure 5);
16. fill the field `List of columns to select` with "1,7,8,4,6" (the red arrow in figure 5);
17. unselect all the other options;
18. execute;
19. set the output file datatype to *bed*;
20. you can rename the output as "final_coverage_depth.bed".

![Figure 4]({{site.url}}{{site.baseurl}}/images/cov_fig4.png)

![Figure 5]({{site.url}}{{site.baseurl}}/images/cov_fig5.png)

### 5.Select positions with low coverage depth
The following procedure can be used to obtain a *bed* file listing base postitions with a coverage depth lower than a certain treshold (for instance 20x).

1. Select the `Filter` tool in the `Filter and Sort` section (indicated by the blue arrow in figure 6);
2. select the file "final_coverage_depth.bed";
2. add the following expression "c5<20" to filter all positions with a coverage depth lower than (green arrow in figure 6)("c5" stands for the fifth column, in this case reporting the coverage depth);
4. excute.

![Figure 6]({{site.url}}{{site.baseurl}}/images/cov_fig6.png)

The output file, which will be called "low_coverage_depth.bed" from now on, will only list all the positions with a depth lower than 20x.

### 6.Merge low coverage regions
If you want to merge the positions with low coverage depth in larger genomic intervals to be used for further analyses (i.e.: Sanger sequencing of regions not covered by your NGS experiment), you may want to use the `bedtools MergeBED` tool in the `Operate on Genomic Intervals` section (see the blue arrow in figure 7). 

1. Select the file "low_coverage_depth.bed";
2. set the maximum distance between features (i.e.: the different positions listed in your file) allowed for features to be merged (green arrow in figure 7): if it is 0 only overlapping and/or book-ended features are merged, while if it is set to 10 (or any other different positive integer of your choice), features at the maximum distance of 10 bases will be merged; 
3. if you want, you may apply some operations to certain columns to get further information in your output file. For instance you may:
	1. click on "Insert Applying operations to columns from merged intervals" (red arrow in figure 7), 
	2. specify the column on which you want to perform the operation (in this case column 5), 
	3. and select the operation from the drop-down list(in this case "Min", which calculates the minimun value of coverage depth among all the positions that will be merged in a single interval) (yellow arrow in figure 7);
4. you may add as many operations as you need. In this example we will also calculate the maximum value of coverage depth;
5. execute.	

![Figure 7]({{site.url}}{{site.baseurl}}/images/cov_fig7.png)

The output file will have the following fileds (columns): chromosome, starting and ending positions of low coverage regions, the minimum and the maximum coverage depth in each region.

   ----------
   **Warning**: please be aware that the columns to use for calculations may be different compared to the example here considered, depending on the amount of columns of your *bed* files.
   
   **Warning**: please be aware that the Galaxy preview of your file shows a header row that does not properly define columns in your files (it is just a standard header for the UCSC bed format).
   
   ----------

>#### Final notes
>The procedures listed above are are to be taken as examples of the possible operations that can be performed on bed files with bedtools (you may check out their website to get further information: [BEDtools](https://bedtools.readthedocs.io/en/latest/content/bedtools-suite.html)) ad text manipulation tools available on Galaxy.
>
>Furthermore, please be aware that the tool `bedtools Compute both the depth and breadth of coverage` does not perform any filtering based on read quality: if your are interested in that aspect you may want to rely on different tools. 
