---
layout: page
title: Genomic databases and variant annotation
summary: "Genomic databases and variant annotation"
sidebar_link: true
---

# Variant annotation

-	Once called, variants (SNPs and InDels) need to be annotated
-	We want to know for example if a variant is located in a gene, if itâ€™s in the coding portion of that gene, if it causes an aminoacid substitution, if that substitution is deleterious for the encoded protein function.
-	**Variant annotation** is the process of attaching biological information from multiple available source (**public databases**) to variants
---
### Gene model

-	The choice of gene model is essential for variant downstream variant annotation: it describe genomic positions of genes and each exon-intron exact locations
-	Different gene models can give different annotations:
-	![Tool interface]({{site.url}}{{site.baseurl}}/images/brca1var.jpg)

- RefSeq [RefSeq](https://www.ncbi.nlm.nih.gov/refseq/)
- Ensembl [Ensembl] (https://www.ensembl.org/Homo_sapiens/Info/Index)
- Gencode [Gencode] (https://www.gencodegenes.org/human/)
---
### Sequence variant nomenclature

-	Variant nomenclature should be described univocally:
- HGVS [HGVS] (https://varnomen.hgvs.org/)
- HGMC [HGMC] (https://www.genenames.org/)

---
### Variant class
   - Sequence Ontology [Sequence ontology] (http://www.sequenceontology.org/)
   - ![Tool interface]({{site.url}}{{site.baseurl}}/images/seqOnt.png)

---
### Population sequencing db
   - GnomAD [gnomAD] (https://gnomad.broadinstitute.org/)
   - ExAC [ExAC] (http://exac.broadinstitute.org/)
   - 1000 Genomes Project [1000 genomes] (https://www.internationalgenome.org/)
   - NHLBI-ESP 6500 exomes [6500 exomes] (https://evs.gs.washington.edu/EVS/)
   - dbSNP [Single Nucleotide Polymorphisms] (https://www.ncbi.nlm.nih.gov/snp/)

---
### Variant-disease/gene-disease db

   - HGMD [Human Gene Mutation Database] (http://www.hgmd.cf.ac.uk/ac/index.php)
   - Clinvar [clinically relevant variants known in a gene] (https://www.ncbi.nlm.nih.gov/clinvar/)
   - Simple ClinVar [simple ClinVar] (http://simple-clinvar.broadinstitute.org/)
   - OMIM [Online Mendelian Inheritance in Man] (https://www.omim.org/)

---
### Annotation Software and tools

   - Web interface
   - VEP: [Variant Effect Predictor] (https://www.ensembl.org/info/docs/tools/vep/index.html)
   - VAAST [Variant Annotation, Analysis and Search Tool] ()
   - ANNOVAR [Annovar]( http://annovar.openbioinformatics.org/en/latest/)
   - SNPEff [Single Nucleotide Polymorphism Effect] (http://snpeff.sourceforge.net/)
   - KGGSeq [KGGSeq] (http://grass.cgs.hku.hk/limx/kggseq/)
