---
layout: page
title: Genomic databases and variant annotation
summary: "Genomic databases and variant annotation"
sidebar_link: true
---

# Variant annotation

Once called, variants (SNPs and InDels) need to be annotated

We want to know for example if a variant is located in a gene, if it’s in the coding portion of that gene, if it causes an aminoacid substitution, if that substitution is deleterious for the encoded protein function.

**Variant annotation** is the process of attaching biological information from multiple available source (**public databases**) to variants

---
### Gene model

The choice of gene model is essential for variant downstream variant annotation: it describe genomic positions of genes and each exon-intron exact locations

Different gene models can give different annotations:
- ![BRCA1]({{site.url}}{{site.baseurl}}/images/brca1_var.jpg)

**Figure 1.** Variant indicated by the red dashed line can be annotated as *intronic* or *exonic* (on one of the UCSC transcript variants), depending on the adopted gene model

- [RefSeq](https://www.ncbi.nlm.nih.gov/refseq/)
- [Ensembl](https://www.ensembl.org/Homo_sapiens/Info/Index)
- [Gencode](https://www.gencodegenes.org/human/)

---
### Sequence variant nomenclature

Variant nomenclature should be described univocally:

- [HGVS](https://varnomen.hgvs.org/)
- [HGMC](https://www.genenames.org/)

---
### Variant class

- [Sequence Ontology](http://www.sequenceontology.org/)
- ![Sequence Ontology]({{site.url}}{{site.baseurl}}/images/seqOnt.png)

---
### Population sequencing db

- [gnomAD](https://gnomad.broadinstitute.org/)
- [ExAC](http://exac.broadinstitute.org/)
- [1000 Genomes Project](https://www.internationalgenome.org/)
- [NHLBI-ESP 6500 exomes](https://evs.gs.washington.edu/EVS/)
- [dbSNP](https://www.ncbi.nlm.nih.gov/snp/)

---
### Variant-disease/gene-disease db

- [Human Gene Mutation Database](http://www.hgmd.cf.ac.uk/ac/index.php)
- [clinically relevant variants known in a gene](https://www.ncbi.nlm.nih.gov/clinvar/)
- [Simple ClinVar](http://simple-clinvar.broadinstitute.org/)
- [Online Mendelian Inheritance in Man](https://www.omim.org/)

---

- [**dbNSFP**](https://sites.google.com/site/jpopgen/dbNSFP), a big database of curated annotations and precomputed functional predictions for all potential non-synonymous and splice-site single-nucleotide variants in the human genome

---
### Annotation Software and tools

- [Variant Effect Predictor](https://www.ensembl.org/info/docs/tools/vep/index.html)
- [ANNOVAR]( http://annovar.openbioinformatics.org/en/latest/)
- [Single Nucleotide Polymorphism Effect](http://snpeff.sourceforge.net/)
- [KGGSeq](http://grass.cgs.hku.hk/limx/kggseq/)

- Web Interface:
   - [wAnnovar](http://wannovar.wglab.org)
   - [VEP](http://grch37.ensembl.org/Homo_sapiens/Tools/VEP)

---
### Annotation and filtering with wANNOVAR

- The web tool [wANNOVAR](http://wannovar.wglab.org/index.php) allows for rapid annotation of your variants and for some basic filtering steps to find disease genes.
- It is based on its command line counterpart [ANNOVAR](http://annovar.openbioinformatics.org/), but it is more user-friendly since it does not require any programming skills.
- The output consist in tabular text files that can be easily manipulated with Excel or other spreadsheet programs.
- The annotation is performed against some of the most commonly used databases: RefSeq, UCSC Known, ENSEMBL/Genecode, dbSNP, ClinVar, 1000genomes, ExAC, ESP6500, gnomAD (minor allele frequencies in different populations are reported) and various precalculated prediction scores for any possible single nucleotide variant in coding regions (see [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP)). 
- The gene-based annotation results in a single row for each input variant: only the most deleterious consequence is reported (i.e.: if a certain variant may result to be missense for one transcript and nonsense for a second transcript, only the latter consequence will be reported).
- Unfortunately, unlike the command line version, wANNOVAR does neither allow for the use of custom annotation databases, nor for the selection of different pubblicly available databases.
- To annotate your file with wANNOVAR you need to provide your email address, to be notified when the annotation is complete,  and just upload your input file (or paste a series of variant in the designated field).
- ![Figure 1]({{site.url}}{{site.baseurl}}/images/wann_fig1.png)
- Results are usually ready within minutes.
- You will get both *csv* files or *txt* files (which can be saved as they are by clicking on the specific link, right-clicking in any point of the page and selecting `Save as`). They can be both opened with Excel or other spreadsheet programs (concerning *csv* files, please ensure that comma are set as default list separator/delimiter in your version of Excel).
- You will also get both `exome summary results` (only conding variants are included) and `genome summary results` (all variants included).
- You can also provide a list of Disease or Phenotype Terms that the program can use for filtering your results (only for single sample `vcf` files).
- ![Figure 2]({{site.url}}{{site.baseurl}}/images/wann_fig2.png)

Finally, there are some Parameter Settings that can be modified:
   - Result duration: it can now only be set to "1 day", since your files will be automatically removed after 24 hours.
   - Reference genome: you can choose between hg19 (GRCh37) and hg38 (GRCh38).
   - Input Format: you can upload not only *vcf* files, but also other kinds of variant files.
   - Gene Definition: the database you want to use for gene function annotation. Three oprtions are available: RefSeq, UCSC Known, ENSEMBL/Gencode
   - Individual analysis: the "Individual analysis" option allows you to perform further filtering steps (based on the Disease/Phenotype terms or the Disease Model) on a single sample (if you upload a multisample *vcf* only the first sample will be considered); the "All Annotaions" option will annotate all variants in your multisample *vcf*, maintaining the original columns of your *vcf* as the last columns of your output file.
   - Disease Model: this options allows for some basic filtering of your variants based on the expected mechanism of inheritance. In mainly consider frequencies, sample genotypes and consequences at level of genic function. FIltering step are summarized among the results and they are only performed on a single sample: the program does not perform any multisample evaluation (i.e.: variant segregation in a trio) and cannot classify any variant as de novo even if you provide a multisample *vcf* with parental genotypes.

---
### Clinical databases for further manual variant annotation

- Once you have obtained a file with the annotation of your variants, you might find useful to annotate also the involved genes, in order to know, for instance, the list of diseases that may be associated with them.

- Some databases that can the jb are the following:
1. The gene2phenotype dataset ([G2P](https://www.ebi.ac.uk/gene2phenotype/disclaimer)) integrates data on genes, variants and phenotypes for example relating to developmental disorders. In the "Download" section you will find both databases of genes related to cancer and to developmental disorders. Those files report for each gene listed: the OMIM number, the associated disease name, the disease OMIM number, the disease category (you can fing more details in the "Terminology" section), whether the disease is caused by biallelic or monoallelic variants ("allelic requirement"), the expected category of variant to be causative of the disease, and a few other details.
2. In the "Download" section of the [OMIM database](https://www.omim.org/downloads/), if you register for research and educational use, you may obtain different lists of OMIM genes and their associated phenotypes.
3. Among the files available for download from the [gnomAD database](https://gnomad.broadinstitute.org/downloads#constraint), you may get per-gene constraint scores (for further details, please check the paper by the Exome Aggregation Consortium on "Nature. 2016 Aug 18; 536(7616):285–291."). Those score may indicate if a specific gene is expected to be intolerant to loss-of-function variants (pLI) (haploinsufficiency), or if it is predicted to be associate to recessive diseases.
- In the end, you can add these annotations to your wANNOVAR files by using the `VLOOKUP` function in Excel.
