---
layout: page
title: Copy number variants
summary: "Identification of Copy Number Variants from targeted sequencing data"
sidebar_link: true
---

# Copy Number Variants

- *Copy Number Variants* (CNVs) are imbalances in the genomic material of an individual that result in either DNA **deletions** (copy loss) or **duplications**
- *CNVs* can cause/predispose to human diseases by altering gene structure/dosage or by **position effect**
- *CNV* size ranges from 50 bp to megabases
- Classical methods to identify *CNVs* use array-based technologies (SNP/CGH)
- Computational approaches have been developed to identify *CNVs* from targeted sequencing data

## CNV detection from targeted sequencing data

### Computational approaches

- There are four main methods for *CNV* identification from short-read +NGS data (see figure below):
  - **Read Count** (RC)
  - **Read Pair** (RP)
  - **Split Read** (SR)
  - **De Novo Assembly** (AS)
  
- **RP** and **SR** require continuous coverage of the *CNV* region or reads encompassing *CNV* breakpoints, as in whole genome sequencing. The sparse nature and small size of exonic targets hamper the application of these methods to targeted sequencing. 
- **RC** is the best suited method for *CNV* detection from whole exomes or gene panels where:
  - deletions appear as exonic targets devoid of reads
  - duplications appear as exonic targets characterized by excess of coverage

### 

[![Galaxy]({{site.url}}{{site.baseurl}}/images/methods_identification_cnv.png)]
Figure 1. Methods for detection of CNVs in short read NGS data (adapted from Tattini et al., 2015)
---

