---
layout: page
title: Copy number variants
summary: "Identification of Copy Number Variants from targeted sequencing data"
sidebar_link: true
---

# Copy Number Variants

- *Copy Number Variants* (CNVs) are imbalances in the genomic material of an individual that result in either DNA **deletions** (copy loss) or **duplications**
- *CNVs* can cause/predispose to human diseases by altering gene structure/dosage or by **position effect**
- *CNV* size ranges from 50 bp to megabases
- Classical methods to identify *CNVs* use array-based technologies (SNP/CGH)
- Computational approaches have been developed to identify *CNVs* in targeted sequencing data from **hybrid capture** experiments

## CNV detection from targeted sequencing data

### Computational approaches

- There are four main methods for *CNV* identification from short-read +NGS data (see figure below):
  - **Read Count** (RC)
  - **Read Pair** (RP)
  - **Split Read** (SR)
  - **De Novo Assembly** (AS)
  
- *RP* and *SR* require continuous coverage of the *CNV* region or reads encompassing *CNV* breakpoints, as in whole genome sequencing. The sparse nature and small size of exonic targets hamper the application of these methods to targeted sequencing. 
- *RC* is the best suited method for *CNV* detection from whole exomes or gene panels where:
  - deletions appear as exonic targets devoid of reads
  - duplications appear as exonic targets characterized by excess of coverage

[![CNV detection methods]({{site.url}}{{site.baseurl}}/images/methods_identification_cnv.png)]
**Figure 1**. Methods for detection of CNVs in short read NGS data (adapted from [Tattini et al., 2015](https://doi.org/10.3389/fbioe.2015.00092))

### Data normalization

In targeted sequencing, a method to study DNA copy number variation by *RC* is to consider the **exon mean read count** (EMRC):

*EMRC = RCe/Le*

where *RCe* is the number of reads aligned to a target genomic region e and *Le* is the size of that same genomic region in base pairs (Magi et al., 2013).
Three major bias sources are known to affect *EMRC* dramatically in targeted sequencing data:
- local **GC content** percentage
- genomic **mappability**
- target region **size**
These biases contribute to non uniform read depth across target regions and, together with the sparse target nature, challenge the applicability of *RC* methods to targeted data.
In this analysis we compared the
distribution and the CNV prediction capability for both
single-sample EMRC data and the ratio between EMRC
data from two samples.
The histograms in Figure 1g show that for single-sample
data (with the median normalized to copy number two),
the EMRC distributions for genomic regions with different
DNA copy number states have a significant overlap
and completely fail to predict the exact number of copies,
as shown in Figure 1i, where the Pearson correlation coefficient
calculated between the real and predicted DNA
copy number values is R = 0.19. On the other hand, the
EMRC ratio between two samples allows for a better discrimination
of genomic regions with different numbers
of DNA copies, as illustrated in Figures 1h and 1j, where
the Pearson correlation coefficient between the real and
predicted DNA copy number values is R = 0.80. Remarkably,
as shown in Figure 1j, normalized ERMC ratios can
distinguish between even intermediate CN ratios, such as
2/3, 3/4, 4/5 and 3/2, 4/3, 4/2, despite their overlapping
distributions. For these reasons, in all the analyses we performed
for this work, we decided to use the ratio between
EMRC data from test and control samples to identify
genomic regions involved in CNVs: in particular, we chose
to use the log-transformed ratio (log2 ratio) between test
and control samples normalized with the LOWESS scatter
plot normalization procedure (see Additional file 1 for
more details).


[![Data Normalization]({{site.url}}{{site.baseurl}}/images/normalization_EMRC.png)]
**Figure 2**. Methods for detection of CNVs in short read NGS data (adapted from [Magi et al., 2013](http://genomebiology.com/2013/14/10/R120))

---
