---
layout: page
title: Variants/genes annotation and prioritization
summary: "Variants/genes annotation and prioritization"
sidebar_link: true
---

# Variants/genes annotation and prioritization


- Once annotated, variants need to be filtered and prioritized
  - WES: Tens of thousands
  - WGS: Millions

# No universal filters, they depend on the experimental features

## Variant impact

- First of all you usually want to filter variants by consequence on the encoded protein, keeping those which have an higher impact on protein:
  - Missense
  - Nonsense
  - Splice sites
  - Frameshift indels
  - Inframe indels

## Variant frequency
 
- Common variants are unlikely associated with a clinical condition
- A rare variant will probably have a higher functional effect on the protein
- Frequency cut-off have to be customized on each different case
- Typical cut-offs: 1% - 0.1%
- Allele frequencies may differ a lot between different populations

## Variant effect prediction Tools

- Tools that predict consequences of amino acid substitutions on protein function
- They give a score and/or a prediction in terms of "Tolerated", "Deleterious" (SIFT) or "Probably Damaging", "Possibly Damaging", "Benign" (Polyphen2)
  - fitCons
  - GERP++
  - SIFT
  - PolyPhen2
  - CADD
  - DANN
  - Condel
  - fathmm
  - MutationTaster
  - MutationAssessor
  - REVEL

# ACMG/AMP 2015 guidelines

- 28 evaluation criteria for the clinical interpretation of variants
  - Criteria falls into 3 sets:
     - pathogenic/likely pathogenic (P/LP)
     - benign/likely benign (B/LB)
     - variant of unknown significance (VUS)

- Intervar[http://wintervar.wglab.org]: software for automatically interpretation of the 28 criteria


## Prioritization
